grove/tomography/utils.py

```diff
@@ -0,0 +1,640 @@
+
+        data = self.basis_transform * tm * self.basis_transform.H
+        sop = qt.Qobj(data, dims=[self.ops[0].dims, self.ops[0].dims])
+        sop.superrep = "super"
+        return sop
+
+    def __str__(self):
+        return "<span[{}]>".format(",".join(self.labels))
+
+    def __eq__(self, other):
+        return (self.labels == other.labels and all(
+            [(my_op - o_op).norm('fro') < EPS for (my_op, o_op) in zip(self.ops, other.ops)]))
+
+
+PAULI_BASIS = OperatorBasis(
+    [("I", QI / np.sqrt(2)), ("X", QX / np.sqrt(2)),
+     ("Y", QY / np.sqrt(2)), ("Z", QZ / np.sqrt(2))])
+
+
+def gell_mann_basis(qudit_dim):
+    """
+    Compute an orthonormal hermitian operator basis for the space of observables of a qudit with
+    `d` levels. These are identical to the Gell-Mann matrices up to normalization.
+    See also this `article <https://arxiv.org/pdf/quant-ph/0301152.pdf>`_ .
+
+    :param int qudit_dim: The dimension of the underlying Hilbert space.
+    :return: The normalized Gell-Mann operator basis.
+    :rtype: OperatorBasis
+    """
+
+    labels_ops = [('I', qt.qeye(qudit_dim) / np.sqrt(qudit_dim))]
+    xops = []
+    yops = []
+
+    # off diagonals
+    for jj in range(qudit_dim):
+        for kk in range(jj):
+            sigma_jk = qt.projection(qudit_dim, jj, kk)
+            xops.append(('X{}{}'.format(kk, jj), (sigma_jk + sigma_jk.dag()) / np.sqrt(2)))
+            yops.append(('Y{}{}'.format(kk, jj), (sigma_jk - sigma_jk.dag()) / np.sqrt(2) * 1j))
+
+    labels_ops += xops + yops
+
+    # diagonals
+    for jj in range(1, qudit_dim):
+        diag = np.array([1.] * jj + [-jj] + [0.] * (qudit_dim - jj - 1))
+        diag /= np.linalg.norm(diag, 2)
+        labels_ops.append(('Z{}'.format(jj),
+                           qt.qdiags([diag], [0], [[qudit_dim], [qudit_dim]],
+                                     (qudit_dim, qudit_dim))))
+
+    return OperatorBasis(labels_ops)
+
+
+def n_qubit_pauli_basis(n):
+    """
+    Construct the tensor product operator basis of `n` PAULI_BASIS's.
+
+    :param int n: The number of qubits.
+    :return: The product Pauli operator basis of `n` qubits
+    :rtype: OperatorBasis
+    """
+    if n >= 1:
+        return PAULI_BASIS ** n
+    else:  # pragma no coverage
+        raise ValueError("n = {} should be at least 1.".format(n))
+
+
+def n_qubit_ground_state(n):
+    """
+    Construct the tensor product of `n` ground states |0>.
+
+    :param int n: The number of qubits.
+    :return: The state |000...0> for `n` qubits.
+    :rtype: qutip.Qobj
+    """
+    return qt.tensor(*([GS] * n))
+
+
+def to_density_matrix(state):
+    """
+    Convert a Hilbert space vector to a density matrix.
+
+    :param qt.basis state: The state to convert into a density matrix.
+    :return: The density operator corresponding to state.
+    :rtype: qutip.qobj.Qobj
+    """
+    return state * state.dag()
+
+
+GS = to_density_matrix(qt.basis(2, 0))
```

Comment:
these constants are always seeded in between, we should organize them in a central place, so we don't have to search all over the code base